wandb/agents/pyagent.py

```diff
@@ -85,6 +85,9 @@ def __init__(
         # files = (glob_config, loc_config)
         self._api = InternalApi()
         self._agent_id = None
+        # if the directory to log to is not set, set it
+        if os.environ.get("WANDB_DIR") is None:
+            os.environ["WANDB_DIR"] = os.path.abspath(os.getcwd())
```

Comment:
@raubitsj it seems like we generally aren'y setting env variables on behalf of the user.  Should we be doing this in `wandb.settings`?

Reply:
@vanpelt, i would prefer not to set env vars at all.  That seems like an anti-pattern for passing state around.

we should avoid unless we need something for some compatibility with the old cli that we can start to deprecate.